auth/threelegged.go

```diff
@@ -66,6 +70,31 @@ type Options3LO struct {
 	AuthHandlerOpts *AuthorizationHandlerOptions
 }
 
+func (o *Options3LO) validate() error {
+	if o == nil {
+		return errors.New("auth: options must be provided")
```

Comment:
Would it make sense to add "Options3LO" to these error messages? For example: ""auth: Options3LO. ClientID must be provided", etc?